### DISCUSSION TITLE
WP Themes & Different Primary Colors for Multiple Themes on Subsite Network.

### DISCUSSION THREAD

**Discussion by minemindmedia** [tailwindlabs/tailwindcss#5720] [Help] [answered]
I apologize in advance if this doesn't really belong here but I have a question about how this may or may not be possible. I am working on a multisite WordPress project using the Roots stack (roots.io) and will have about 5 sub-sites all using the same theme. If possible, how could I go about changing the themes primary color for each sub-site without duplicating the theme?

At first I was thinking I would just have to create a new theme for each subsite just to apply a different primary color. Then when I need to update anything else within the theme, I'd have to do it 5 times. The color is the only thing that will change. Plus some images using the WP customizer. 

I was thinking of maybe using the WordPress customizer with a color picker and then just injecting that hex to where I have all my primary colors set.

The other idea I had was to create all of the primary colors in the tailwind config file, then use the WP customizer to select an option for the colors I setup in the config and then apply the selected color to the elements via php conditionals. This may be the best way to achieve what I'm trying to do but I'm wondering if I am missing another option that may even be better.

I also apologize if this doesn't make enough sense.

**Comment by simonswiss** [comment, marked as answer]
Hey!

That's a very good question. One thing I think off straight away is to use CSS variables here. You could set your primary color in a CSS file, defined in a CSS variable.

Then, you can use that variable in your theme definition. 

This would also open up cool possibilities like previewing a different color that updates when you select it with the color picker (which you'd use to update the variable color value).

Here's a screencast that shows how to use CSS variables with Tailwind CSS colours:

https://www.youtube.com/watch?v=MAtaT8BZEAo

<img width="1301" alt="image" src="https://user-images.githubusercontent.com/485747/136300364-1cef9492-96f8-414a-a823-d5eeafa5c002.png">

---

Another approach you could take is use a `preset`, which would be your "base" config/theme, and then each theme would add the primary color on top of that preset:

https://tailwindcss.com/docs/presets

Again, here's another screencast on how to use presets:

https://www.youtube.com/watch?v=AaZhspn7-MM

<img width="774" alt="image" src="https://user-images.githubusercontent.com/485747/136300509-94787de7-c2f0-4d12-ae9b-b842862cd909.png">

Some of the syntax/APIs on these videos may be slightly outdated by now, but they'll both definitely give you a good idea on how to achieve what you're trying to do.

Hope it helps!

**Comment by minemindmedia** [comment]
@simonswiss -

Thank you so much for getting back to me about this!
Amazing! I will have a look into everything and hopefully get this working!

Kindly-

**Reply by simonswiss** [reply]
No problem, glad it was useful 👍

**Reply by minemindmedia** [reply]
Hi @simonswiss --

I've finally got around to watching your screencast and implementing this. Everything works great when I include the "theme-one", "theme-two", or "theme-three" classes manually to the body class or parent containers, etc.

The issue is when I use the WordPress customizer to select a different option from a dropdown list. It defaults to the ":root" styles. I believe it may need to be compiled on my dev environment, which is why that doesn't work. Any thoughts on this?

Here's my setup:

`  function theme_skin($wp_customize){
 
    //add section
    $wp_customize->add_section( 'theme_skin_section', array(
 
                'title' => 'Theme Skin',
                'priority' => 10
    ));
 
    //add setting
    $wp_customize->add_setting( 'theme_skin', array(
                'default' => '',
    ));
 
    //add control
    $wp_customize->add_control( 'theme_skin_control', array(
                'label' => 'Select Skin',
                'type'  => 'select',
                'section' => 'theme_skin_section',
                'settings' => 'theme_skin',
			'choices' => array(
				'default' => 'Default',
				'one' => 'One',
				'two' => 'Two',
				'three' => 'Three',
			)
    ));
 
}
 
add_action( 'customize_register', 'theme_skin' );`

> @layer base {
>     :root {
>       --color-text-base: 0, 0, 0;
>       --color-text-muted: 84, 84, 84;
>       --color-text-inverted: 255, 255, 255;
>       --color-fill: 247, 247, 247;
>       --color-button-accent: 0, 0, 0;
>       --color-button-accent-hover: 56, 56, 56;
>       --color-button-muted: 255, 255, 255;
>     }
>     .theme-one {
>       --color-text-base: 255, 255, 255;
>       --color-text-muted: 254, 202, 202;
>       --color-text-inverted: 220, 38, 38;
>       --color-fill: 185, 28, 28;
>       --color-button-accent: 255, 255, 255;
>       --color-button-accent-hover: 254, 242, 242;
>       --color-button-muted: 239, 68, 68;
>     }
>     .theme-two {
>       --color-text-base: 17, 24, 2;
>       --color-text-muted: 47, 67, 6;
>       --color-text-inverted: 235, 250, 204;
>       --color-fill: 179, 255, 23;
>       --color-button-accent: 36, 52, 3;
>       --color-button-accent-hover: 55, 79, 5;
>       --color-button-muted: 212, 255, 122;
>     }
>     .theme-three {
>         --color-text-base: 17, 24, 2;
>         --color-text-muted: 47, 67, 6;
>         --color-text-inverted: 235, 250, 204;
>         --color-fill: 179, 255, 23;
>         --color-button-accent: 36, 52, 3;
>         --color-button-accent-hover: 55, 79, 5;
>         --color-button-muted: 212, 255, 122;
>     }
> }
> 

```
module.exports = {
  mode: 'jit',
  purge: {
    content: [
      './app/**/*.php',
      './resources/**/*.php',
      './resources/**/*.js',
    ],
  },
  theme: {
    extend: {
      textColor: {
        skin: {
          base: withOpacity('--color-text-base'),
          muted: withOpacity('--color-text-muted'),
          inverted: withOpacity('--color-text-inverted'),
        },
      },
      backgroundColor: {
        skin: {
          fill: withOpacity('--color-fill'),
          'button-accent': withOpacity('--color-button-accent'),
          'button-accent-hover': withOpacity('--color-button-accent-hover'),
          'button-muted': withOpacity('--color-button-muted'),
        },
      },
      gradientColorStops: {
        skin: {
          hue: withOpacity('--color-fill'),
        },
      },
    },
    colors: {
      primary: colors.red,
      transparent: 'transparent',
      current: 'currentColor',
      black: colors.black,
      white: colors.white,
      gray: colors.trueGray,
      indigo: colors.indigo,
      red: colors.red,
      rose: colors.rose,
      amber: colors.amber,
      orange: colors.orange,
      yellow: colors.yellow,
      lime: colors.lime,
      green: colors.green,
      emerald: colors.emerald,
      teal: colors.teal,
      cyan: colors.cyan,
      lightBlue: colors.lightBlue,
      blue: colors.blue,
      indigo: colors.indigo,
      violet: colors.violet,
      purple: colors.purple,
      pink: colors.pink
    },
  },
  variants: {
    extend: {},
  },
  plugins: [
    require('@tailwindcss/forms'),
  ],
}
```

The class is then added to parent containers via:
`theme-<?php esc_html_e( get_theme_mod( 'theme_skin' ) ); ?>`

**Reply by minemindmedia** [reply]
Upon further review, if I select "theme-three" from the WP customizer, it appears to apply those neon colors.
If I change the customizer to "theme-one" or "theme-two", it applies the ":root" colors.

**Reply by minemindmedia** [reply]
Further investigation shows it doesn't apply the new colors at all once I deploy to my staging server/env.

**Reply by minemindmedia** [reply]
I looked at my app.css file, where my scss is compiled and I only see the :root set. It doesn't appear to be compiling the class sets for "theme-one", "theme-two" or "theme-three".

Only shows:

```
:root {
    --color-text-base: 0, 0, 0;
    --color-text-muted: 84, 84, 84;
    --color-text-inverted: 255, 255, 255;
    --color-fill: 247, 247, 247;
    --color-button-accent: 0, 0, 0;
    --color-button-accent-hover: 56, 56, 56;
    --color-button-muted: 255, 255, 255;
  }
```

**Reply by minemindmedia** [reply]
Further investigation shows that if I name my groups with a colon, like :root, then they show up when compiled but of course they don't work as class names for my parent elements:

![image](https://user-images.githubusercontent.com/8854361/139146755-ec086ac9-430e-4d53-914e-e89ac497ed49.png)

**Comment by minemindmedia** [comment]
@simonswiss --

I was able to get around my issue by removing the: @layer base {}
Not sure how that may affect anything.

Can you help me understand how I would set colors for borders?

I have this:

```
.theme-two {
  --color-text-base: 17, 24, 2;
  --color-text-muted: 47, 67, 6;
  --color-text-inverted: 235, 250, 204;
  --color-fill: 179, 255, 23;
  --color-border: 179, 255, 23;
  --color-button-accent: 36, 52, 3;
  --color-button-accent-hover: 55, 79, 5;
  --color-button-muted: 212, 255, 122;
}
```

```
      borderColor: {
        skin: {
          base: withOpacity('--color-border'),
        },
      },
```

Lastly, I have this input to add a border on focus:

`<input x-ref="input" type="search" name="" placeholder="Search" value="" autofocus required class="w-full bg-gray-100 border border-gray-300 focus:border-none placeholder-primary-700 uppercase text-sm tracking-wider h-16 px-5 pr-10 text-xl focus:outline-none pl-12 focus:border focus:border-skin-base">`

It doesn't work and I can't determine what I am doing wrong.

**Comment by minemindmedia** [comment]
I managed to get the borderColor setup and working a while ago but now I'm having trouble getting it setup for ring colors and placeholders...

I've added --color-ring-base below:

```
.theme-two {
    --color-text-base: 1, 132, 69;
    --color-text-dark: 51, 51, 51;
    --color-text-muted: 47, 67, 6;
    --color-text-inverted: 255, 255, 255;
    --color-fill: 1, 132, 69;
    --color-border: 1, 132, 69;
    --color-button-light: 1, 132, 69;
    --color-button-dark: 0, 91, 46;
    --color-ring-base: 1, 132, 69;
}
```

in my tailwind config file I have added:

```
      ringColor: {
        skin: {
          base: withOpacity('--color-ring-base'),
          dark: withOpacity('--color-text-dark'),
        },
      },
```

And for an input I'm calling these classes:

`focus:ring-1 focus:ring-skin-base`

But that doesn't appear to work.
